Module has no attribute "interchange"

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
2. Indicate which type checker you are using (`mypy` or  `pyright`).
3. Show the error message received from that type checker while checking your example.

```console
(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ cat t.py
import pandas as pd

df = pd.DataFrame({'a': [1, 2, 3]})

res = pd.api.interchange.from_dataframe(df)
print(res)

(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ mypy t.py
t.py:5: error: Module has no attribute "interchange"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ pip freeze | grep pandas
pandas==2.0.1
pandas-stubs==2.0.1.230501
(.venv) marcogorelli@DESKTOP-U8OKFP3:~/tmp$ pip freeze | grep mypy
mypy==1.3.0
mypy-extensions==1.0.0
```

**Please complete the following information:**
 - OS: [e.g. Windows, Linux, MacOS] linux
 - OS Version [e.g. 22]
 - python version 3.10.6
 - version of type checker mypy
 - version of installed `pandas-stubs` 2.0.1.230501


**Additional context**
Add any other context about the problem here.

--- a/pandas-stubs/api/__init__.pyi
+++ b/pandas-stubs/api/__init__.pyi
@@ -1,5 +1,6 @@
 from pandas.api import (
     extensions as extensions,
     indexers as indexers,
+    interchange as interchange,
     types as types,
 )

--- a/tests/test_api_types.py
+++ b/tests/test_api_types.py
@@ -470,3 +470,11 @@ def test_check_extension_dtypes() -> None:
     check_ext_dtype(pd.SparseDtype)
     check_ext_dtype(pd.Float32Dtype)
     check_ext_dtype(pd.Float64Dtype)
+
+
+def test_from_dataframe() -> None:
+    # GH 712
+    check(
+        assert_type(pd.api.interchange.from_dataframe(dframe), pd.DataFrame),
+        pd.DataFrame,
+    )

